### ISSUE TITLE
ENH: allow linspace to take 1D vectors as start and end

### ISSUE THREAD

**Issue by SjoerdOptLand** [numpy/numpy#2944]
I have a line segment in _N_-dimensional space, defined by its begin- and endpoints (both 1D vectors). For example in 3D:

```
start = array([0,1,2])
end = array([1,3,6])
```

I would like to be able segment the line in _k_ segments, for example 4, or 5 including the endpoints. The result would then be:

```
 array([[ 0.  ,  0.25,  0.5 ,  0.75,  1.  ],
       [ 1.  ,  1.5 ,  2.  ,  2.5 ,  3.  ],
       [ 2.  ,  3.  ,  4.  ,  5.  ,  6.  ]]) 
```

I extended (and broke) linspace to not only accept scalar, but also 1D vector start- and endpoints:

```
def linspace(start,stop,num=50,endpoint=True,retstep=False):
    spannedRange = stop-start
    numberOfSteps = num - (1 if endpoint else 0)
    step = array(spannedRange/numberOfSteps,ndmin=2).T
    coefficientColumn = numpy.arange(num)
    samples = numpy.kron(coefficientColumn,step)+array(start,ndmin=2).T
    if retstep:
        return (samples,step)
    else:
        return samples
```

Would it be possible (without breaking linspace former behaviour) to incorporate this functionality into NumPy?

**Comment by mindlapse** [comment]
The np.transpose of @seberg's idea would be a great for this API - it would create a list of all the interpolated vector points between two given points

**Comment by eric-wieser** [comment]
If supporting 1D arrays is desirable, we may as well support ND, and add an axis argument calculated with

    axis = np.core.multiarray.normalize_axis_index(axis, broadcast(...).ndim + 1)

**Comment by mhvk** [comment]
Implemented in #12388, so closing.
